src/Validator/ValidatorFactory.php

```diff
@@ -27,9 +27,12 @@ public function createValidator(MetadataFactory $metadataFactory): ValidatorInte
     {
         $builder = Validation::createValidatorBuilder()
             ->setMetadataFactory($metadataFactory)
-            ->setConstraintValidatorFactory($this->constraintValidatorFactory)
         ;
 
+        if (null !== $this->constraintValidatorFactory) {
+            $builder->setConstraintValidatorFactory($this->constraintValidatorFactory);
+        }
```

Comment:
This block doesn't make much sense, because this class is useless without the `ConstraintValidatorFactory` being injected into it. I would rather add an exception into constructor, as I did in the `InputValidator` class, to prevent the use of this class.

In the case on the `$defaultTranslator` it's perfectly fine if no Translator was injected, because the `InputValidator` can work without it. But without the `ConstraintValidatorFactory` it won't work correctly.